Source/WebCore/html/HTMLBodyElement.cpp

```diff
@@ -115,25 +115,24 @@ const AtomString& HTMLBodyElement::eventNameForWindowEventHandlerAttribute(const
 void HTMLBodyElement::parseAttribute(const QualifiedName& name, const AtomString& value)
 {
     if (name == vlinkAttr || name == alinkAttr || name == linkAttr) {
-        if (value.isNull()) {
-            if (name == linkAttr)
+        auto parsedColor = parseLegacyColorValue(value);
+        if (name == linkAttr) {
+            if (parsedColor)
+                document().setLinkColor(*parsedColor);
```

Comment:
We should consider refactoring later and just have set functions on the document class that take an optional rather than pairs of set and reset functions. Also noting that since `Color` has an invalid value we could consider using that instead of `std::optional`. I guess that’s what `WTF::Markable` is for?

Reply:
Yeah, that's probably a good refactoring to do afterwards.